nixos/modules/system/boot/networkd.nix

```diff
@@ -2812,9 +2812,15 @@ let
 
       environment.etc."systemd/networkd.conf" = renderConfig cfg.config;
 
-      systemd.services.systemd-networkd = {
+      systemd.services.systemd-networkd = let
+        isReloadableUnitFileName = unitFileName: strings.hasSuffix ".network" unitFileName;
+        reloadableUnitFiles = attrsets.filterAttrs (k: v: isReloadableUnitFileName k) unitFiles;
+        nonReloadableUnitFiles = attrsets.filterAttrs (k: v: !isReloadableUnitFileName k) unitFiles;
+        unitFileSources = unitFiles: map (x: x.source) (attrValues unitFiles);
```

Comment:
Maybe it would be better to use [`lib.partition`](https://teu5us.github.io/nix-lib.html#lib.lists.partition) for this?

Reply:
Yeah, seems okay to me